compiler/parser/parser.peg

```diff
@@ -1082,6 +1082,14 @@ Function
       id := &ast.ID{Kind: "ID", Name: "regexp_replace", Loc: ast.NewLoc(c.pos.offset, c.pos.offset+14)}
       return newCall(c, id, []any{arg0, arg1, arg2}, where), nil
     }
+  / !FuncGuard EXTRACT __ "(" __ part:Expr _ FROM _ e:Expr __ ")" where:WhereClause? {
+      return &ast.CallExtract{
+        Kind: "CallExtract",
+        Part: part.(ast.Expr),
+        Expr: e.(ast.Expr), 
+        Loc: loc(c),
+      }, nil
```

Comment:
Tests pass with change this so do we need ast.CallExtract?
```suggestion
      id := &ast.ID{Kind: "ID", Name: "date_part", Loc: ast.NewLoc(c.pos.offset, c.pos.offset+9)}
      return newCall(c, id, []any{part, e}, where), nil      
```

Reply:
My philosophy behind ast Nodes is that you use be able to use it to generally re-create the original textual representation and this is why imho `EXTRACT(PART FROM this)` needs its own ast Node because it doesn't have the same syntax as a standard function call.

Reply:
Is that goal really worth the extra code, especially considering that we already have cases where two language constructs map to the same AST node (e.g., `count(*)` and `count()`, `a ? b : c` and `case when a then b` end`)?

Reply:
If you wanted to write a formatter along the lines of what go has (which is something I'd like to do at some point) then you need to structure your ast this way. I have an aversion to doing hacks like these that came from when I did the pr to tie errors to ast locations as they just made that work more complicated.